src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,114 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 1 << 4;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int size;
+    private int threshold;
+    private int capacity;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+    }
 
     @Override
     public void put(K key, V value) {
-
+        table = resize();
+        int hash = (key == null) ? 0 : key.hashCode();
+        int bucket = Math.abs(hash) % capacity;
```

Comment:
```suggestion
        int hash = (key == null) ? 0 : key.hashCode();
        int bucket = Math.abs(hash) % capacity;
```
I suggest move this code to method. You use it in different palces.

Reply:
ok. I'll check it